### ISSUE TITLE
S3fs version is 1.9.4,I find warning log printf "Not enough local storage to cache write request till multipart upload can start" .what will be affect?

### ISSUE THREAD

**Issue by loongyiyao** [s3fs-fuse/s3fs-fuse#2444]
#### How to run s3fs, if applicable

 /usr/bin/s3fs fangzhenyun /dev/mount1 -o url=https://video.ge.com:9000 -o endpoint=cn-east-1 -o sigv2 -o passwd_file=/home/zxsrtn/.passwd-s3fs -o use_path_request_style -o allow_other -o umask=0 -o use_cache=/dev/cache1 -o del_cache -o ensure_diskfree=12288 -o enable_noobj_cache -o parallel_count=20 -o multipart_size=52 -o dbglevel=warn -o logfile=/home/output.log

```
```

#### s3fs syslog messages (`grep s3fs /var/log/syslog`, `journalctl | grep s3fs`, or `s3fs outputs`)

```
```

### Details about issue
2024-04-13T01:19:54.389Z [WAN] fdcache_entity.cpp:WriteMixMultipart(2307): Not enough local storage to cache write request till multipart upload can start: [path=/vcd001/280/230/28023006/video/10-001328023006-1-20240412212748782-00001-20240413032810.mpg.rtp/119.dat][physical_fd=1036][offset=9407996][size=127492]
2024-04-13T01:19:54.389Z [WAN] s3fs.cpp:s3fs_write(2958): failed to write file(/vcd001/280/230/28023006/video/10-001328023006-1-20240412212748782-00001-20240413032810.mpg.rtp/119.dat). result=-28
2024-04-13T01:19:54.810Z [WAN] fdcache_entity.cpp:WriteMixMultipart(2307): Not enough local storage to cache write request till multipart upload can start: [path=/vcd001/280/231/28023135/video/10-001328023135-1-20240412212740722-00001-20240413032818.mpg.rtp/119.dat][physical_fd=230][offset=11246738][size=127854]
2024-04-13T01:19:54.810Z [WAN] s3fs.cpp:s3fs_write(2958): failed to write file(/vcd001/280/231/28023135/video/10-001328023135-1-20240412212740722-00001-20240413032818.mpg.rtp/119.dat). result=-28
2024-04-13T01:27:29.073Z [ERR] curl.cpp:RequestPerform(2544): HEAD HTTP response code 400, returning EPERM.
2024-04-13T01:37:07.311Z [WAN] fdcache_entity.cpp:WriteMixMultipart(2307): Not enough local storage to cache write request till multipart upload can start: [path=/vcd001/280/228/28022809/video/10-001328022809-1-20240412212736882-00002-20240413092815.mpg.rtp/3.dat][physical_fd=417][offset=1828178][size=129710]
2024-04-13T01:37:07.311Z [WAN] s3fs.cpp:s3fs_write(2958): failed to write file(/vcd001/280/228/28022809/video/10-001328022809-1-20240412212736882-00002-20240413092815.mpg.rtp/3.dat). result=-28

**Comment by ggtakec** [comment]
@loongyiyao 
In order to upload, the cache file (and work file) partition must have free space equal to the size of the file you are trying to upload (more precisely, the size of the updating area size).
Since you have specified `ensure_diskfree=12288`, the `<partition free space size> - <12GB>` will be the size that can be used when uploading.
Since s3fs does not automatically delete cached files, it may be necessary to delete cache files using an external process.

And, there is something a little concerning about your log.
The directory path to create the cache file is specified as `use_cache=/dev/cache1`, but in the error log it is `/vcd001/280/231/...`.
Do you know anything about these directory paths?

**Comment by loongyiyao** [comment]
> @loongyiyao In order to upload, the cache file (and work file) partition must have free space equal to the size of the file you are trying to upload (more precisely, the size of the updating area size). Since you have specified `ensure_diskfree=12288`, the `<partition free space size> - <12GB>` will be the size that can be used when uploading. Since s3fs does not automatically delete cached files, it may be necessary to delete cache files using an external process.

> hi,delete cache will ccause data loss?
> And, there is something a little concerning about your log. The directory path to create the cache file is specified as `use_cache=/dev/cache1`, but in the error log it is `/vcd001/280/231/...`. Do you know anything about these directory paths?
> /dev/cache1 this is cache dictionary, i mount real dictionary is /dev/mount, so do not print /dev/cache1 in s3 logs.

**Comment by ggtakec** [comment]
@loongyiyao 
Please let me check a few things.

> /dev/cache1 this is cache dictionary,

Does this mean the directory is mounted using `cachefs` or something? (I'm not familiar with this.)
The path specified in the s3fs's `use_cache` option is assumed to be a normal directory path, and if it is a path mounted with `cachefs`, I don't think we have confirmed its behavior.

> delete cache will cause data loss? how delete cache

When deleting a file that exists under the cache directory, it is necessary that the file is not open.
As long as it is not open, deleting it should not be a problem.
(It may be difficult to verify that it is not being accessed)

> Another question is i find Memory leakage by https ways with v1.9.4 version.

Memory leak is a serious problem, so please explain in detail.
If possible, it would be helpful if you could create an issue with only the memory leak as an issue.

**Comment by loongyiyao** [comment]
> Memory leak is a serious problem, so please explain in detail. If possible, it would be helpful if you could create an issue with only the memory leak as an issue. (Is that what you have already reported in #2441?)

@ggtakec yes! i find memory leakage, The memory occupied by s3fs has increased from tens of megabytes to a few grams, and within half a month。i
